src/goose/cli/session.py

```diff
@@ -150,23 +154,37 @@ def run(self) -> None:
         """
         message = self.process_first_message()
         while message:  # Loop until no input (empty string).
-            self.notifier.start()
-            try:
-                self.exchange.add(message)
-                self.reply()  # Process the user message.
-            except KeyboardInterrupt:
-                self.interrupt_reply()
-            except Exception:
-                # rewind to right before the last user message
-                self.exchange.rewind()
-                print(traceback.format_exc())
-                print(
-                    "\n[red]The error above was an exception we were not able to handle.\n\n[/]"
-                    + "These errors are often related to connection or authentication\n"
-                    + "We've removed the conversation up to the most recent user message"
-                    + " - [yellow]depending on the error you may be able to continue[/]"
-                )
-            self.notifier.stop()
+            # Start a span with the default tracer. This is present with opentelemetry-instrument
+            with self.tracer.start_as_current_span(message.role) as span:
+                span.set_attribute("goose.role", message.role)
```

Comment:
In practice, it tends to be a minor optimization but as much as possible we would like to pass attributes to the `attributes` parameter of `start_as_current_span` since they can be used in sampling that way

Reply:
good idea, done